server/services/socket.js

```diff
@@ -91,6 +91,13 @@ exports.emitWinner = (roomId, player) => io.to(`room-${roomId}`).emit("winner",
 // When game is over, tell the users
 exports.emitEndGame = (roomId) => io.to(`room-${roomId}`).emit("end");
 
+// Close the room when the game has ended
+exports.closeRoom = (roomId) => {
+    sockets.forEach(socket => socket.handshake.session.roomId === roomId && socket.leave(`room-${roomId}`));
```

Comment:
lets say you had 100 rooms and the first one was the correct roomId, you would still be looping around the other 99 here. You could use 'find' to get the first instance in the array that matches.
`
const socket = sockets.find(socket => socket.handshake.session.roomId === roomId);
socket .leave(`room-${roomId}`));
`
or something similar

Reply:
ah sorry i think i misunderstood, there will be multiple sockets that match the room id, ignore this

Reply:
Yeah, I think we're looping through the sockets here to make sure they leave their rooms when the game ends